specification/modsimworkbench/resource-manager/Microsoft.ModSimWorkbench/preview/2021-03-01-preview/files.json

```diff
@@ -0,0 +1,476 @@
+        "result": {
+          "type": "string",
+          "description": "Request to download files from an ModSim Workbench chamber data pipeline action"
+        }
+      }
+    },
+    "FilePropertiesFormat": {
+      "type": "object",
+      "properties": {
+        "provisioningState": {
+          "$ref": "./modsimworkbenches.json#/definitions/ProvisioningState",
+          "description": "Gets the status of the file at the time the operation was called"
+        },
+        "timestamp": {
+          "type": "string",
+          "format": "date-time",
+          "description": "The modified timestamp of the file."
+        },
+        "dataPipeline": {
+          "type": "string",
+          "description": "The direction of ModSim Workbench chamber data pipeline where the file located."
+        },
+        "requestId": {
+          "type": "string",
+          "description": "The resource ID of ModSim Workbench chamber data pipeline request Works and required only if requested is true."
+        },
+        "filePath": {
+          "type": "string",
+          "description": "The Linux File System path for the ModSim Workbench chamber data pipeline file."
+        },
+        "size": {
+          "type": "integer",
+          "format": "int64",
+          "description": "The size of the file.",
+          "example": 1073741824
+        },
+        "checksum": {
+          "type": "string",
+          "description": "The MD5 hash of the file."
+        }
+      },
+      "description": "File properties."
+    },
+    "File": {
+      "properties": {
+        "properties": {
+          "x-ms-client-flatten": true,
+          "$ref": "#/definitions/FilePropertiesFormat",
+          "description": "File properties",
+          "type": "object"
+        }
+      },
+      "allOf": [
+        {
+          "$ref": "./modsimworkbenches.json#/definitions/Resource"
+        }
+      ],
+      "description": "File resource"
+    },
+    "FilesManageParameters": {
+      "description": "Manage files in an ModSim Workbench chamber data pipeline.",
+      "type": "object",
+      "properties": {
+        "dataPipeline": {
+          "description": "Inbound or outbound of the ModSim Workbench chamber Data pipeline.",
+          "type": "string",
+          "enum": [
+            "Inbound",
+            "Outbound"
+          ],
+          "x-ms-enum": {
+            "name": "dataPipeline",
+            "modelAsString": true
+          }
+        },
+        "action": {
+          "description": "Specifies the file management action.",
+          "type": "string",
+          "enum": [
+            "Delete",
+            "Archive"
+          ],
+          "x-ms-enum": {
+            "name": "action",
+            "modelAsString": true
+          }
+        },
+        "force": {
+          "description": "Specifies whether to force delete. Works if action is Delete",
+          "type": "string",
+          "enum": [
+            "True",
```

Comment:
>"True" [](http://example.com/codeflow?start=12&length=6)

Having an enumeration with 'True' and 'False' values is an antipattern, there should really be some kind of linter rule against this...
I'd suggest either giving it more meaningful enum value names that can't be confused with a boolean, or changing it back to bool

Reply:
Another option would be to get rid of this parameter entirely and just have a separate action 'ForceDelete' to go with 'Delete'

Reply:
Consider also documenting, in what cases is forcing delete helpful, as opposed to just deleting.

Reply:
Previous arm reviewer suggested to change all bool to enum

Reply:
I don't think this should be what they were imagining.